### PR TITLE
Support tuple assignment in inline assembly

### PR DESCRIPTION
**Pull request by axic**
This only support a single right hand side and variable number of left hand side variables.

The analysis checks whether the number of deposited items match the required ones.

### CODE REVIEW

**Comment by axic** [comment]
Do we want to support multiple right hand side items?

**Comment by chriseth** [comment]
I would not want to support "multiple right hand sides". The only reason of existence for this syntactical construct is to handle functions returning multiple values. Introducing tuple expressions would complicated the language.

File: libsolidity/inlineasm/AsmCodeGen.cpp [hunk 1]
```diff
@@ -202,11 +202,16 @@ class CodeTransform: public boost::static_visitor<>
...
-		expectDeposit(1, height);
-		auto& var = boost::get<Scope::Variable>(m_scope.identifiers.at(_varDecl.variable.name));
-		var.stackHeight = height;
-		var.active = true;
+		expectDeposit(expectedItems, height);
+		int index = expectedItems;
+		for (auto const& variable: _varDecl.variables)
+		{
+			auto& var = boost::get<Scope::Variable>(m_scope.identifiers.at(variable.name));
+			var.stackHeight = --index;
```
Comment (chriseth) [inline libsolidity/inlineasm/AsmCodeGen.cpp]: I think this changes semantics. `height` can be much more than just the current expression stack.
The following could work:
```
index--;
var.stackHeight = height - index;
```

**Review by chriseth — CHANGES_REQUESTED** [review]

**Comment by axic** [comment]
This is missing assignment, only support variable declaration.

File: test/libsolidity/InlineAssembly.cpp [hunk 2]
```diff
@@ -161,6 +161,16 @@ BOOST_AUTO_TEST_CASE(vardecl)
...
 }
 
+BOOST_AUTO_TEST_CASE(vardecl_multi)
+{
+	BOOST_CHECK(successParse("{ function f() -> x, y {} let x, y := f() }"));
+}
+
+BOOST_AUTO_TEST_CASE(vardecl_multi_conflict)
+{
+	CHECK_PARSE_ERROR("{ function f() -> x, y {} let x, x := f() }", ParserError, "");
```
Comment (axic) [inline test/libsolidity/InlineAssembly.cpp]: Need failure reason.

**Comment by axic** [comment]
@chriseth happy to merge this?

**Review by chriseth — APPROVED** [review]

**Pull request merged**